tests/test_pytest_plugin.py

```diff
@@ -769,3 +771,39 @@ def test_outsource():
     assert project.storage(storage_dir) == snapshot(
         ["2cf24dba5fb0a30e26e83b2ac5b9e29e1b161e5c1fa7425e73043362938b9824.html"]
     )
+
+
+def test_find_pyproject_in_parent_directories(tmp_path, monkeypatch):
+    from inline_snapshot.pytest_plugin import pytest_configure
+    from inline_snapshot import _config
+
+    config_dir = tmp_path / "config"
+    config_dir.mkdir()
+    pyproject_upper = tmp_path / "pyproject.toml"
+    pyproject_upper.touch()
+    pyproject_lower = config_dir / "pyproject.toml"
+    pyproject_lower.touch()
+
+    config = MagicMock()
+    config.rootpath = tmp_path / "config"
+
+    class EarlyBubble(Exception):
+        def __init__(self, value):
+            self.value = value
+
+    def read_config(path):
+        raise EarlyBubble(path)
+
+    monkeypatch.setattr(_config, "read_config", read_config)
+
+    try:
+        pytest_configure(config)
+    except EarlyBubble as bubble:
+        assert bubble.value == pyproject_lower
+
+    pyproject_lower.unlink()
+
+    try:
+        pytest_configure(config)
+    except EarlyBubble as bubble:
+        assert bubble.value == pyproject_upper
```

Comment:
We could also unlink the upper pyproject and assert that it climbs up to the root (`/pyproject.toml`).